### PR TITLE
utils_test: Nested virtualization support for Avocado-VT

### PR DESCRIPTION
**Pull request by balamuruhans**
add support to run `avocado-vt` inside guest that can be used to
bringup nested guests from tests.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -4423,9 +4423,9 @@ def get_block(self, p_dict={}):
         block_list = self.process_info_block(blocks_info)
         for block in block_list:
             for key, value in six.iteritems(p_dict):
-                    # for new qemu we just deal with key = [removable,
-                    # file,backing_file], for other types key, we should
-                    # fixup later
+                # for new qemu we just deal with key = [removable,
+                # file,backing_file], for other types key, we should
```
Comment (sathnaga) [inline virttest/qemu_vm.py]: this change is not relevant here...send as separate PR.

File: virttest/utils_test/__init__.py [hunk 2]
```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
...
-        cmd = "cd %s;[ -d %s ] || git clone %s" % (test_path,
-                                                   testrepo.split("/")[-1].split('.')[0],
-                                                   testrepo)
-        status, output = session.cmd_status_output(cmd, timeout=100)
-        if status > 0:
-            raise exceptions.TestError("Downloading test failed: %s" % output)
+        if self.avocado_vt:
+            cmd = "avocado vt-bootstrap --vt-type %s" % self.vt_type
+            if self.guest_image:
+                cmd += " --vt-guest-os %s --yes-to-all" % self.guest_image
```
Comment (sathnaga) [inline virttest/utils_test/__init__.py]: move `--yes-to-all`  to above cmd= statement so that even without self.guest_image we go by default downloading ...

File: virttest/utils_test/__init__.py [hunk 3]
```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
...
+            status, output = self.session.cmd_status_output(cmd, timeout=100)
```
Comment (sathnaga) [inline virttest/utils_test/__init__.py]: this timeout can be increased to few mins as we might have large guest image to download and uncompress

File: virttest/utils_test/__init__.py [hunk 4]
```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
...
+        cmd = "avocado run %s"
+        for test_each in self.testlist:
+            avocado_cmd = str(cmd)
+            if self.avocado_vt:
+                avocado_cmd %= test_each.strip()
+                if self.vt_type:
+                    avocado_cmd += " --vt-type %s" % self.vt_type
+                if self.vt_only_filter:
+                    avocado_cmd += " --vt-only-filter %s" % self.vt_only_filter
+                if self.vt_type:
```
Comment (sathnaga) [inline virttest/utils_test/__init__.py]: it should be `if  self.vt_no_filter:`

File: virttest/utils_test/__init__.py [hunk 5]
```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
...
+                                            "https://github.com/avocado-framework/avocado.git")
+        self.plugins = {'pip': ['avocado-framework-plugin-varianter-yaml-to-mux', 'avocado-framework-plugin-result-html'],
+                        'package': [],
+                        'git': ['varianter_yaml_to_mux', 'html']}
+        self.prerequisites = {'git': ['git'], 'python': ['python', 'python-pip'], 'python2': ['python2', 'python2-pip']}
+        if self.avocado_vt:
+            self.vt_type = self.params.get("vt_type", "libvirt")
+            self.vt_only_filter = self.params.get("vt_only_filter", "")
+            self.vt_no_filter = self.params.get("vt_no_filter", "")
+            self.guest_image = self.params.get("vt_guest_image", "")
```
Comment (sathnaga) [inline virttest/utils_test/__init__.py]: prefix all params with test specific tags as this confuse with L0 params...
e-g:- `avocadoguesttest_vt_type`

File: virttest/utils_test/__init__.py [hunk 6]
```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
...
-            cmd = "avocado run %s --job-results-dir %s --job-timeout %d" % (
-                testcase, result_path, timeout)
-            if mux:
-                cmd += " -m %s" % mux
-            if add_args:
-                cmd += " %s" % add_args
-            status, output = session.cmd_status_output(cmd, timeout=timeout)
-            if status > 0:
+        cmd = "avocado run %s"
+        for test_each in self.testlist:
```
Comment (sathnaga) [inline virttest/utils_test/__init__.py]: for avocado-vt at this moment there is no test specific filters or args passed unlike mux file in misc tests so we need not iterate and run avocado run multiple times, rather we can run all tests together...make it simple at present..

Comment (sathnaga) [reply virttest/utils_test/__init__.py]: I understand it might helpful to report per test status, but avocado-vt already have a good per-test view even we run multiple tests together....

File: virttest/utils_test/__init__.py (hunk 5 above)
Comment (balamuruhans) [reply virttest/utils_test/__init__.py]: right, I am handling it in different way for differentiating with Level and guest specific params.

**Comment by balamuruhans** [comment]
@sathnaga thank you for review, I have made the changes as suggested.

**Comment by balamuruhans** [comment]
@sathnaga I have rebased the code, I have tested it and posted the results in https://github.com/avocado-framework/avocado-vt/pull/1947#issuecomment-460536102.
